lib/daemon_runner/semaphore.rb

```diff
@@ -22,6 +22,7 @@ def lock(name, limit = 3, **options)
         if block_given?
           begin
             lock_thr = semaphore.renew
+            loop until semaphore.locked?
```

Comment:
I think this should be above `semaphore.renew` and if `locked?` is `false` you need to call `try_lock` to try to obtain the lock again.